WebDriverAgentLib/Categories/XCUIElement+FBUID.m

```diff
@@ -43,6 +48,18 @@ + (void)load
   NSAssert(XCElementSnapshotCls != nil, @"Could not locate XCElementSnapshot class");
   Method uidMethod = class_getInstanceMethod(self.class, @selector(fb_uid));
   class_addMethod(XCElementSnapshotCls, @selector(fb_uid), method_getImplementation(uidMethod), method_getTypeEncoding(uidMethod));
+  
+  // Support for Xcode 14.3 requires disabling the new predicate validator, see https://github.com/appium/appium/issues/18444
+  Class XCTElementQueryTransformerPredicateValidatorCls = objc_lookUpClass("XCTElementQueryTransformerPredicateValidator");
+  if (XCTElementQueryTransformerPredicateValidatorCls != nil) {
+    Method validatePredicateMethod = class_getClassMethod(XCTElementQueryTransformerPredicateValidatorCls, NSSelectorFromString(@"validatePredicate:withExpressionsAllowed:"));
+    if (validatePredicateMethod != nil) {
+      IMP swizzledImp = (IMP)swizzled_validatePredicateWithExpressionsAllowed;
+      method_setImplementation(validatePredicateMethod, swizzledImp);
+    } else {
+      [FBLogger log:@"Could not find method +[XCTElementQueryTransformerPredicateValidator validatePredicate:withExpressionsAllowed:]"];
+    }
```

Comment:
[minor] I think that comment meant:

```suggestion
    if (validatePredicateMethod == nil) {
      [FBLogger log:@"Could not find method +[XCTElementQueryTransformerPredicateValidator validatePredicate:withExpressionsAllowed:]"];
    } else {
      IMP swizzledImp = (IMP)swizzled_validatePredicateWithExpressionsAllowed;
      method_setImplementation(validatePredicateMethod, swizzledImp);
    }
```

Reply:
the above condition could also be flipped and turned to a guard, e.g.

```
if (...) {
 big block of code here
}
```

is harder to read in comparison to

```
if (!...) {
  return;
}

big block of code
```